libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -544,6 +610,51 @@ void AP_ESC_Telem::update()
                 _last_telem_log_ms[i] = telemdata.last_update_ms;
                 _last_rpm_log_us[i] = rpmdata.last_update_us;
             }
+
+#if EDTV2_SUPPORT
+            // Write an EDTv2 message, if there is any update
+            uint16_t edt2_status = telemdata.edt2_status;
+            uint16_t edt2_stress = telemdata.edt2_stress;
+            if (EDT2_HAS_NEW_DATA(edt2_status | edt2_stress)) {
+                // Could probably be faster/smaller with bitmasking, but not sure
+                uint8_t status = 0;
+                if (EDT2_HAS_NEW_DATA(edt2_stress)) {
+                    status |= uint8_t(log_Edt2_Status::HAS_STRESS_DATA);
+                }
+                if (EDT2_HAS_NEW_DATA(edt2_status)) {
+                    status |= uint8_t(log_Edt2_Status::HAS_STATUS_DATA);
+                }
+                if (EDT2_ALERT_BIT_FROM_STATUS(edt2_status)) {
+                    status |= uint8_t(log_Edt2_Status::ALERT_BIT);
+                }
+                if (EDT2_WARNING_BIT_FROM_STATUS(edt2_status)) {
+                    status |= uint8_t(log_Edt2_Status::WARNING_BIT);
+                }
+                if (EDT2_ERROR_BIT_FROM_STATUS(edt2_status)) {
+                    status |= uint8_t(log_Edt2_Status::ERROR_BIT);
+                }
+                // An EDT2 status message is:
+                //   id: starts from 0
+                //   stress: the current stress which comes from edt2_stress
+                //   max_stress: the maximum stress which comes from edt2_status
+                //   status: the status bits which come from both
+                const struct log_Edt2 pkt_edt2{
+                    LOG_PACKET_HEADER_INIT(uint8_t(LOG_EDT2_MSG)),
+                    time_us     : now_us64,
+                    instance    : i,
+                    stress      : EDT2_STRESS_FROM_STRESS(edt2_stress),
+                    max_stress  : EDT2_STRESS_FROM_STATUS(edt2_status),
+                    status      : status,
+                };
+                if (AP::logger().WriteBlock_first_succeed(&pkt_edt2, sizeof(pkt_edt2))) {
+                    // Only clean the telem_updated bits if the write succeeded.
+                    // This is important because, if rate limiting is enabled,
+                    // the log-on-change behavior may lose a lot of entries
```

Comment:
I think using `WriteBlock` bypasses the rate limiting? In anycase it does seem reasonable to check so we don't loose information. Maybe we need to add the return value to one of the methods that does rate limiting.

Reply:
`WriteBlock` that writes ESC messages above does not bypass rate limiting.
`WriteBlock_first_succeed` does the same but returns a boolean.

Here [24-05-04_14-39-33_edt_rate_limited.bin.zip](https://github.com/ArduPilot/ardupilot/files/15210016/24-05-04_14-39-33_edt_rate_limited.bin.zip) is an example of log with 5Hz rate limiting, and one can see that EDT2 messages come at 5Hz too.

Reply:
Your right it does look like it should limit correctly. However, your also getting messages at between 5 and 12 hz.

![image](https://github.com/ArduPilot/ardupilot/assets/33176108/c573e9be-f833-4a52-ac1c-65b942c240d5)

Reply:
5Hz with four motors should be 20Hz, so I guess its the new data check that is actually slowing it down?

Reply:
I meant (at most) 5Hz per each motor of course, just the same what happens with the ESC messages.

Reply:
Thanks for the explanation.